sys/src/libc/port/reallocarray.c

```diff
@@ -28,6 +28,22 @@
 void *
 reallocarray(void *optr, size_t nmemb, size_t size)
 {
+	/* If both size or nmemb are 0, and realloc is called with
```

Comment:
I'd shorten the comment:

    /* malloc(0) returns a valid non-nil pointer, while realloc(p, 0) is equivalent to free(p). */